sdks/python/apache_beam/io/iobase.py

```diff
@@ -1238,128 +1235,38 @@ def try_claim(self, position):
     raise NotImplementedError
 
 
-class ThreadsafeRestrictionTracker(object):
```

Comment:
Could you split out a separate commit that moves these classes from the ones that add new logic?

Reply:
Are you suggesting that create a new PR to move existing class like `ThreadsafeRestrictionTracker` and make this PR focus on watermark estimator only? If that's the case, I would prefer letting Chad's [PR](https://github.com/apache/beam/pull/10593) check in first. I can do the refactor for `ThreadsafeRestrictionTracker` related stuff on the top of typing changes.

Reply:
Finished rebasing on the top of master branch,
Filed https://issues.apache.org/jira/browse/BEAM-9296 for adding type check.